src/lib/checkbox/checkbox-foundation.ts

```diff
@@ -1,209 +1,194 @@
 import { ICustomElementFoundation } from '@tylertech/forge-core';
 
 import { ICheckboxAdapter } from './checkbox-adapter';
-import { CHECKBOX_CONSTANTS } from './checkbox-constants';
+import { CHECKBOX_CONSTANTS, CheckboxLabelPosition, CheckboxState } from './checkbox-constants';
 
 export interface ICheckboxFoundation extends ICustomElementFoundation {
+  checked: boolean;
+  defaultChecked: boolean;
+  indeterminate: boolean;
+  value: string;
   dense: boolean;
+  disabled: boolean;
+  required: boolean;
+  readonly: boolean;
+  labelPosition: CheckboxLabelPosition;
+  syncValidity(hasCustomValidityError: boolean): void;
+  setValidity(flags?: ValidityStateFlags | undefined, message?: string | undefined): void;
 }
 
 export class CheckboxFoundation implements ICheckboxFoundation {
+  // State
+  private _checked = false;
+  private _defaultChecked = false;
+  private _indeterminate = false;
+  private _value = 'on';
   private _dense = false;
-  private _currentCheckState = CHECKBOX_CONSTANTS.strings.TRANSITION_STATE_INIT;
-  private _currentAnimationClass = '';
-  private _animEndLatchTimer = 0;
-  private _enableAnimationEndHandler = false;
+  private _disabled = false;
+  private _required = false;
+  private _readonly = false;
+  private _labelPosition: CheckboxLabelPosition = 'end';
 
-  private _handleChange!: () => void;
-  private _inputAttributeChangedListener: (name: string, value: string) => void;
-  private _handleAnimationEnd: () => void;
+  private get _submittedValue(): string | null {
+    return this._checked ? this._value : null;
+  }
+  private get _formState(): CheckboxState {
+    return this._checked
+      ? this._indeterminate
+        ? 'checked-indeterminate'
+        : 'checked'
+      : this._indeterminate
+        ? 'unchecked-indeterminate'
+        : 'unchecked';
```

Comment:
Should we move these strings to constants?

Reply:
I thought they made more sense as string literals 🤷‍♂️